src/game/scenes/lobby.c

```diff
@@ -539,145 +553,35 @@ void lobby_entered_name(component *c, void *userdata) {
-                      settings_get()->net.net_listen_port_start, end_port);
-                if(settings_get()->net.net_listen_port_start == 0) {
-                    address.port = rand_int(65535 - 1024) + 1024;
-                    randtries++;
-                    if(randtries > 10) {
-                        DEBUG("Failed to initialize ENet server, could not allocate random port");
-                        return;
-                    }
-                } else {
-                    address.port++;
-                    if(address.port > end_port) {
-                        DEBUG("Failed to initialize ENet server, port range exhausted");
-                        return;
-                    }
-                    randtries++;
-                    if(randtries > 10) {
-                        DEBUG("Failed to initialize ENet server, could not allocate port between %d and %d after 10 "
-                              "tries",
-                              settings_get()->net.net_listen_port_start, end_port);
-                        return;
-                    }
-                }
-            }
-        }
-
-        DEBUG("bound to port %d", address.port);
-
-        int ext_port = settings_get()->net.net_ext_port_start;
-        if(ext_port == 0) {
-            // try to use the internal port first
-            ext_port = address.port;
-        }
-        randtries = 0;
-
-        if(nat->type != NAT_TYPE_NONE) {
-            while(!nat_create_mapping(nat, address.port, ext_port)) {
-                if(settings_get()->net.net_ext_port_start == 0) {
-                    ext_port = rand_int(65535 - 1024) + 1024;
-                    randtries++;
-                    if(randtries > 10) {
-                        ext_port = 0;
-                        break;
-                    }
-                } else {
-                    ext_port++;
-                    if(ext_port > settings_get()->net.net_ext_port_end) {
-                        ext_port = 0;
-                        break;
-                    }
-                }
-            }
-        }
-        enet_socket_set_option(local->client->socket, ENET_SOCKOPT_REUSEADDR, 1);
-
-        ENetEvent event;
-        ENetPeer *peer = NULL;
-        enet_address_set_host(&lobby_address, "lobby.openomf.org");
-        // enet_address_set_host(&address, "127.0.0.1");
-        lobby_address.port = 2098;
-        DEBUG("server address is %d", lobby_address.host);
-        /* Initiate the connection, allocating the two channels 0 and 1. */
-        local->peer = enet_host_connect(local->client, &lobby_address, 2, 0);
+        strncpy_or_truncate(local->name, textinput_value(c), sizeof(local->name));
 
-        if(local->peer == NULL) {
-            DEBUG("No available peers for initiating an ENet connection.\n");
+        char version[15];
+        // support git version when not on a tag
+        snprintf(version, 14, "%d.%d.%d", V_MAJOR, V_MINOR, V_PATCH);
+        version[14] = 0;
```

Comment:
smells like something that could be sliced to a utility function ? I think we do this elsewhere too.

Reply:
See #872